### PR TITLE
Feature/355 upload images to blob storage

### PR DESCRIPTION
**Pull request by rodrigo-baldiviezo-jalafund**
[AB#355](https://dev.azure.com/funjaladev/DevLevel3/_boards/board/t/DevLevel3%20Team/Stories/?workitem=355)

# Overview

Upload images to blob storages

### CODE REVIEW

File: src/components/projects/Entry.jsx [hunk 1]
```diff
@@ -64,7 +78,7 @@ function Entry() {
   const hasErrorDescription =
     !!formik.touched.description && !!formik.errors.description;
   const hasErrorName = !!formik.touched.name && !!formik.errors.name;
-  const hasErrorLogo = !!formik.touched.logo && !!formik.errors.logo;
+  // const hasErrorLogo = !!formik.touched.logo && !!formik.errors.logo;
```
Comment (pedro-lima-jalafund) [inline src/components/projects/Entry.jsx]: it is this comment ok?

**Review by niver-copa-jalafund — DISMISSED** [review]

File: src/components/projects/Entry.jsx (hunk 1 above)
Comment (rodrigo-baldiviezo-jalafund) [reply src/components/projects/Entry.jsx]: done

**Review by pedro-lima-jalafund — DISMISSED** [review]

**Review by niver-copa-jalafund — DISMISSED** [review]

File: src/components/projects/Entry.jsx [hunk 2]
```diff
@@ -12,11 +13,22 @@ import "./Entry.scss";
...
 
+  const onFileChange = (event) => {
+    setFileSelected(event.target.files[0]);
+  };
+
+  const onFileUpload = async () => {
+    const fileName = await uploadFileToBlob(fileSelected);
+    setFileSelected(null);
+    return fileName;
+  };
```
Comment (martin-callejas-jalafund) [inline src/components/projects/Entry.jsx:35]: no error handling for this guy?

File: src/components/projects/Entry.jsx [hunk 3]
```diff
@@ -30,13 +42,15 @@ function Entry() {
...
-      const { contact, description, logo, name, textInvitation } = values;
+      const { contact, description, name, textInvitation } = values;
+      let logo;
       const members = memberList.map((member) => {
         return {
           idResume: member.idResume,
           name: member.name,
         };
       });
+      logo = await onFileUpload();
```
Comment (martin-callejas-jalafund) [inline src/components/projects/Entry.jsx:58]: what if this fails? will the save process be broken? @jorge-lema-jalafund

File: src/storage/blobStorage.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  const blobClient = containerClient.getBlockBlobClient(fileName);
+  const options = { blobHTTPHeaders: { blobContentType: file.type } };
+  await blobClient.uploadBrowserData(file, options);
+  return fileName;
+}
+
+export default async function uploadFileToBlob(file) {
+  const blobService = new BlobServiceClient(`${baseUrl}/?${sasToken}`);
+  const containerClient = blobService.getContainerClient(CONTAINER);
+  return await createBlobInContainer(containerClient, file);
```
Comment (martin-callejas-jalafund) [inline src/storage/blobStorage.js:20]: don't we need error handling? CC @jorge-lema-jalafund ?

**Review by martin-callejas-jalafund — COMMENTED** [review]
Looks like it's going to work, but I have concerns about the error handling for this new Azure Blob Storage 3rd party library. Mind taking  a look?

File: src/components/projects/Entry.jsx (hunk 3 above)
Comment (jorge-lema-jalafund) [reply src/components/projects/Entry.jsx:58]: it should exit this function if it failed

File: src/storage/blobStorage.js (hunk 4 above)
Comment (jorge-lema-jalafund) [reply src/storage/blobStorage.js:20]: yes, depending on what we get from the server, we should return something to let the upper layers know if the execution failed or not. So the upper layers would be in charge of showing the error to the UI or proceed to continue with the call to our API from resumes or projects

File: src/components/projects/Entry.jsx (hunk 2 above)
Comment (jorge-lema-jalafund) [reply src/components/projects/Entry.jsx:35]: should see if the result doesn't have an error, if there's an error then show alert to let the user know about the error.

**Review by jorge-lema-jalafund — CHANGES_REQUESTED** [review]

**Review by jorge-lema-jalafund — DISMISSED** [review]
approved

File: src/components/projects/Entry.jsx (hunk 2 above)
Comment (rodrigo-baldiviezo-jalafund) [reply src/components/projects/Entry.jsx:35]: done

**Review by niver-copa-jalafund — DISMISSED** [review]

**Review by pedro-lima-jalafund — DISMISSED** [review]

File: src/components/projects/Entry.jsx (hunk 3 above)
Comment (rodrigo-baldiviezo-jalafund) [reply src/components/projects/Entry.jsx:58]: done

File: src/storage/blobStorage.js (hunk 4 above)
Comment (rodrigo-baldiviezo-jalafund) [reply src/storage/blobStorage.js:20]: done

**Review by martin-callejas-jalafund — APPROVED** [review]

**Review by niver-copa-jalafund — APPROVED** [review]

**Review by jorge-lema-jalafund — APPROVED** [review]
approved

**Pull request merged**